pkg/cli/cmds/agent.go

```diff
@@ -14,7 +14,7 @@ type Agent struct {
 	TokenFile                string
 	ClusterSecret            string
 	ServerURL                string
-	APIAddressCh             chan string
+	APIAddressCh             chan []string
```

Comment:
I am not sure what is the purpose of using slice channel instead of one string, since we are assigning the first taken address anyways? https://github.com/k3s-io/k3s/pull/5143/files#diff-13754aaab77c60dbdff7e6035115d4a091d5a844eff91c70a60c153ebad85dceR383